my-code/src/components/LikeButton/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Container, Button } from './styles';
+import iconHeartWhite from './assets/icon-heart-white.svg';
+import iconHeartGrey from './assets/icon-heart-grey.svg';
+import iconHeartFull from './assets/icon-heart-full.svg';
+
+export default function LikeButton({ imdbID, showText, alwaysVisible }) {
+  const [liked, setLiked] = useState(false);
+  const dispatch = useDispatch();
+
+  // Redux state: get the favourites list.
+  const likes = useSelector(state => state.likesReducer);
+
+  // Based on the favourites list, set the liked state.
+  useEffect(() => {
+    setLiked(likes.indexOf(imdbID) >= 0);
+  }, []);
+
+  /**
+   * Handles the click event on the like button.
+   * @param {*} e
+   */
+  function handleClick(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    setLiked(!liked);
+
+    // Update the favourites list in redux state.
+    dispatch({
+      type: 'SET_LIKE',
+      imdbID,
+    });
+  }
+
+  return (
+    <Container>
+      <Button
+        type="button"
+        className={`heart-button ${liked || alwaysVisible ? 'visible' : ''} ${liked ? 'liked' : ''} ${showText && liked ? 'liked-red' : ''}`}
```

Comment:
You could pass props to the component instead of creating a lot of className rules, it seems a little illegible. 
See more [here](https://www.styled-components.com/docs/basics#passed-props)

Reply:
Done! Much better now.